Automattic/simplenote-macos
# In App Account Deletion

### Fix
fixes #998 

This PR adds in app account deletion to Simplenote Mac.  
In this PR I have also refactored all of the service remotes to use a base Remote class.

<img src="https://cdn-std.droplr.net/files/acc_593859/MxS6Rw" />

### Test
Test account deletion:
1. log into Simplenote with an account you want to delete
2. Go to the Simplenote menu -> Account deletion
3. An alert will appear, test cancel and deletion
4. After confirming the account deletion you will see another alert with a follow up message telling you to check your email
5. a message should appear in your email with instructions to delete the account.
6. Try the same steps again while in airplane mode to ensure an error appears

Other Remotes:
To ensure that the refactored remote classes work:
1. run the unit tests
2. Log out of simplenote and signup for a new account
3. follow all of the steps to confirm you receive the correct emails and an account is setup.

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in 414ecd5f with:
> 
> >- In app account deletion: Now you can delete your account from the app #1013
